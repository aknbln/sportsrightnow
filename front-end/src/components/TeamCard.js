import React, { useEffect } from "react";

import Card from "react-bootstrap/Card";
import Button from "react-bootstrap/Button";
import { Navigate, useNavigate, useSearchParams } from "react-router-dom";

const TeamCard = (props) => {
  const { name, logo, city, league, totalWins, totalLosses, id } = props.sportsTeamData;

  const navigate = useNavigate()

  function ButtonClicked(id) {
    navigate("/teams/instance?id=" + id)
  }

  return (
    <Card
    style= {{backgroundColor: 'lightgray'}}>
      <Card.Img variant="top" src={logo} style={{width: '100%', height: '20vw', objectFit: 'cover'}}/>
      <Card.Body>
        <Card.Title style = {{fontSize: '3vh', color: '#333232'}}>{name}</Card.Title>
        <Card.Text style = {{fontSize: '2vh', color: '#404040'}}>League: {league}</Card.Text>
        <Card.Text style = {{fontSize: '2vh', color: '#404040'}}>City: {city}</Card.Text>
        <Card.Text style = {{fontSize: '2vh', color: '#404040'}}>Wins: {totalWins}</Card.Text>
        <Card.Text style = {{fontSize: '2vh', color: '#404040'}}>Losses: {totalLosses}</Card.Text>
      </Card.Body>
      <Card.Footer style= {{backgroundColor: 'dimgray'}}>
        <Button class="btn btn-primary stretched-link" onClick={() => ButtonClicked(id)}
        style= {{backgroundColor: '#3d405b', borderColor: 'black'}}>
          More Info
        </Button>
      </Card.Footer>
    </Card>
  );
};

export default TeamCard;