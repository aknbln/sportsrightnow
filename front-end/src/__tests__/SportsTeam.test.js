// Replace useNavigates with a dummy function as it doesn't work without a <Router>

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
   ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

import renderer from 'react-test-renderer'
import TeamCard from '../components/TeamCard'
import img from '../assets/images/question.png'

const Team = 
{
    name: "Buffalo Ballers",
    logo: img,
    league: "NBA",
    totalWins: "999",
    totalLosses: "-1",
    city: "Buffalo",
    team_id: "999", 
}

it('TeamCard data correct', () => {
    const component = renderer.create(<TeamCard sportsTeamData={Team}/>)
    expect(component.country = "USA")
})

it('TeamCard renders correctly', () => {
    const component = renderer.create(<TeamCard sportsTeamData={Team}/>)
    let json = component.toJSON()
    expect(json).toMatchSnapshot()
})