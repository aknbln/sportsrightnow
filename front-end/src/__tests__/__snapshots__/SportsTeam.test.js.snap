// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamCard renders correctly 1`] = `
<div
  className="card"
  style={
    Object {
      "backgroundColor": "lightgray",
    }
  }
>
  <img
    className="card-img-top"
    src="question.png"
    style={
      Object {
        "height": "20vw",
        "objectFit": "cover",
        "width": "100%",
      }
    }
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
      style={
        Object {
          "color": "#333232",
          "fontSize": "3vh",
        }
      }
    >
      Buffalo Ballers
    </div>
    <p
      className="card-text"
      style={
        Object {
          "color": "#404040",
          "fontSize": "2vh",
        }
      }
    >
      League: 
      NBA
    </p>
    <p
      className="card-text"
      style={
        Object {
          "color": "#404040",
          "fontSize": "2vh",
        }
      }
    >
      City: 
      Buffalo
    </p>
    <p
      className="card-text"
      style={
        Object {
          "color": "#404040",
          "fontSize": "2vh",
        }
      }
    >
      Wins: 
      999
    </p>
    <p
      className="card-text"
      style={
        Object {
          "color": "#404040",
          "fontSize": "2vh",
        }
      }
    >
      Losses: 
      -1
    </p>
  </div>
  <div
    className="card-footer"
    style={
      Object {
        "backgroundColor": "dimgray",
      }
    }
  >
    <button
      class="btn btn-primary stretched-link"
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#3d405b",
          "borderColor": "black",
        }
      }
      type="button"
    >
      More Info
    </button>
  </div>
</div>
`;
